File: client-scalaz/src/main/scala/org/openapitools/client/api/ProductResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ProductResponse._

case class ProductResponse (
  brand: Option[String],
productType: Option[String],
condition: Option[String],
isbn: Option[String],
asin: Option[String],
catalogNumbers: Option[List[String]],
department: Option[String],
features: Option[List[String]],
minimumPrice: Option[Double],
width: Option[Double],
height: Option[Double],
depth: Option[Double],
weight: Option[Double],
unit: Option[String],
studio: Option[String],
parentalRating: Option[String],
publishDate: Option[Long],
availabilityDate: Option[Long],
size: Option[SizeGroupResponse])

object ProductResponse {
  import DateTimeCodecs._

  implicit val ProductResponseCodecJson: CodecJson[ProductResponse] = CodecJson.derive[ProductResponse]
  implicit val ProductResponseDecoder: EntityDecoder[ProductResponse] = jsonOf[ProductResponse]
  implicit val ProductResponseEncoder: EntityEncoder[ProductResponse] = jsonEncoderOf[ProductResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/product_response.clj
(ns sirqul-io-t-platform.specs.product-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.size-group-response :refer :all]
            )
  (:import (java.io File)))


(def product-response-data
  {
   (ds/opt :brand) string?
   (ds/opt :productType) string?
   (ds/opt :condition) string?
   (ds/opt :isbn) string?
   (ds/opt :asin) string?
   (ds/opt :catalogNumbers) (s/coll-of string?)
   (ds/opt :department) string?
   (ds/opt :features) (s/coll-of string?)
   (ds/opt :minimumPrice) float?
   (ds/opt :width) float?
   (ds/opt :height) float?
   (ds/opt :depth) float?
   (ds/opt :weight) float?
   (ds/opt :unit) string?
   (ds/opt :studio) string?
   (ds/opt :parentalRating) string?
   (ds/opt :publishDate) int?
   (ds/opt :availabilityDate) int?
   (ds/opt :size) size-group-response-spec
   })

(def product-response-spec
  (ds/spec
    {:name ::product-response
     :spec product-response-data}))
